src/appleseed/renderer/kernel/intersection/embreescene.cpp

```diff
@@ -0,0 +1,518 @@
+
+    // Retrieve the object.
+    Object& object = object_instance.get_object();
+
+    // TODO: remove regions.
+    // Retrieve the region kit of the object.
+    const Access<RegionKit> region_kit(&object.get_region_kit());
+
+    assert(region_kit->size() == 1);
+    const size_t region_idx = 0;
+    const IRegion* region = region_kit->at(region_idx);
+    const Access<StaticTriangleTess> tess(&region->get_static_triangle_tess());
+    const size_t motion_steps_count = tess->get_motion_segment_count() + 1;
+    geometry_data.m_motion_steps_count = motion_steps_count;
+
+    //
+    // Retrieve per vertex data.
+    //
+    const size_t vertices_count = tess->m_vertices.size();
+    geometry_data.m_vertices_count = vertices_count;
+    geometry_data.m_vertices_stride = sizeof(GVector3);
+
+    // Allocate memory for the vertices. Keep one extra vertex for padding.
+    geometry_data.m_vertices = new GVector3[vertices_count * motion_steps_count + 1];
+
+    for (size_t i = 0; i < vertices_count; ++i)
+    {
+        const GVector3 vertex_os = tess->m_vertices[i];
+        geometry_data.m_vertices[i] = transform.point_to_parent(vertex_os);
+    }
+
+    // Retrieve assembly space vertices.
+    for (size_t m = 1; m < motion_steps_count; ++m)
+    {
+        for (size_t i = 0; i < vertices_count; ++i)
+        {
+            const GVector3 vertex_os = tess->get_vertex_pose(i, m);
+            geometry_data.m_vertices[vertices_count * m + i] = transform.point_to_parent(vertex_os);
+        }
+    }
+
+    //
+    // Retrieve per primitive data.
+    //
+    const size_t primitives_count = tess->m_primitives.size();
+
+    geometry_data.m_primitives = new uint32[primitives_count * 3];
+    geometry_data.m_primitives_stride = sizeof(uint32) * 3;
+    geometry_data.m_primitives_count = primitives_count;
+
+    for (size_t i = 0; i < primitives_count; ++i)
+    {
+        geometry_data.m_primitives[i * 3] = tess->m_primitives[i].m_v0;
+        geometry_data.m_primitives[i * 3 + 1] = tess->m_primitives[i].m_v1;
+        geometry_data.m_primitives[i * 3 + 2] = tess->m_primitives[i].m_v2;
+    }
+};
+
+inline void collect_curve_data(
+    const ObjectInstance&   object_instance,
+    GeometryData&           geometry_data)
+{
+    //const Transformd& transform = object_instance.get_transform();
+
+    switch(geometry_data.m_geometry_type)
+    {
+    case RTC_GEOMETRY_TYPE_FLAT_BEZIER_CURVE:
+        //
+        // [Note: Girish] Retrieve data from CurveObject here and push it to geometry_data.
+        //
+        
+        break;
+
+    default:
+        assert(!"Unsupported geometry type.");
+        break;
+    }
+}
+
+// Returns minimal tnear needed to compensate double to float transition of ray fields.
+inline float get_min_tnear(
+    const float dir_x,
+    const float dir_y,
+    const float dir_z,
+    const float org_x,
+    const float org_y,
+    const float org_z)
+{
+    float max_dir = max(abs(dir_x), abs(dir_y));
```

Comment:
`foundation/math/minmax.h` provides 3-arg variants that you can use here for clarity.